File: Golang/217. Contains Duplicate.go
func containsDuplicate(nums []int) bool {
    numsDict := make(map[int]int, len(nums))
    for _, num := range nums {
        numsDict[num]++
    }
    for num := range numsDict {
        if numsDict[num] > 1 {
            return true
        }
    }
    return false
}

File: Python3/217. Contains Duplicate.py
class Solution:
    def containsDuplicate(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        # numsDict = {} # 140ms
        # for num in nums:
        #     numsDict[num] = numsDict.get(num, 0) + 1
        # for num in numsDict:
        #     if numsDict[num] > 1:
        #         return True
        # return False
        
        # numsDict = collections.Counter(nums) # 56ms
        # for value in numsDict.values():
        #     if value > 1:
        #         return True
        # return False
        
        return len(nums) != len(set(nums)) # 52ms
